### DISCUSSION TITLE
Java Serialization/deserialization without Kafka

### DISCUSSION THREAD

**Discussion by eskhisov** [Apicurio/apicurio-registry#4767] [Q&A]
Hello,
We want to use registry to serialize/deserialize AVRO schemas in Java projects but we are not using Kafka.  Is it possible?

**Comment by carlesarnal** [comment]
Hello @eskhisov,

In Registry 2 no, it's not possible. For Registry 3, we're working on making the serialization technology messaging technology agnostic so it can be used not only with Kafka but also with other technologies like e.g. Pulsar.

**Reply by EricWittmann** [reply]
This is the right answer.  Although I would say that it is **possible** - just not convenient.  We do not have client serializer/deserializer libraries for non-Kafka use cases.  That's what we are working on for 3.x, as @carlesarnal mentioned.

However, we do have client libraries for accessing Registry via its REST API.  And the Avro serializer/deserializer functionality is coming from standard Avro libraries.  So if you're willing to put those things together yourself, it shouldn't be too hard of a lift.  We just don't have a single library that will do it all for you..... yet.  :)
